packages/workspace/src/browser/workspace-commands.ts

```diff
@@ -213,16 +233,17 @@ export class WorkspaceCommandContribution implements CommandContribution {
                     dialog.open().then(name => {
                         if (name) {
                             const fileUri = parentUri.resolve(name);
-                            this.fileSystem.createFile(fileUri.toString()).then(() => {
-                                open(this.openerService, fileUri);
+                            this.fileSystem.createFile(fileUri.toString()).then(async () => {
+                                await open(this.openerService, fileUri);
```

Comment:
do we need await here or `this.fireCreateNewFile(fileUri)` should go before?

Reply:
Moved `this.fireCreateNewFile(fileUri)` before the `open`. File is already created so the event should be fired first.